### PR TITLE
chore: tweak json-schema tooling, schemas, and test cases

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

This is in preparation for the actual ExperimentConfig objects.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
The generated code is not in this commit yet, but I will add it to pass tests at the end (they pass locally after make gen).

File: schemas/expconf/v0/hyperparameter.json [hunk 1]
```diff
@@ -24,13 +24,6 @@
                 "unionKey": "const:type=categorical",
                 "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameter-categorical.json"
             },
-            {
```
Comment (stoksc) [inline schemas/expconf/v0/hyperparameter.json:27]: What was this for?

**Review by stoksc — APPROVED** [review]
lgtm

File: master/pkg/schemas/defaults.go [hunk 2]
```diff
@@ -62,6 +62,7 @@ func FillDefaults(obj interface{}) {
 // fillDefaults is the recursive layer under FillDefaults.  fillDefaults will return the original
 // input value (not a copy of the original value).
 func fillDefaults(obj reflect.Value, defaultBytes []byte, name string) reflect.Value {
+	// fmt.Printf("fillDefaults on %v (%T)\n", name, obj.Interface())
```
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply master/pkg/schemas/defaults.go:65]: Honestly, they're so useful for debugging I left them there on purpose.

But its not conventional so I'm fine removing them.

File: master/Makefile [hunk 3]
```diff
@@ -12,6 +12,7 @@ clean: ungen
 .PHONY: ungen
 ungen:
 	rm -f $(GENERATED)
+	git co -- $(GENERATED_COMMITTED)
```
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply master/Makefile:15]: oh yeah... I use this so often I don't even realize this is an alias.

File: schemas/expconf/v0/hyperparameter.json (hunk 1 above)
Comment (rb-determined-ai) [reply schemas/expconf/v0/hyperparameter.json:27]: Nested hyperparameters.  Once upon a time I thought the experiment config project was so quick and easy I'd just "throw nested hp in too".

File: master/pkg/schemas/defaults.go [hunk 4]
```diff
@@ -143,16 +150,17 @@ func fillDefaults(obj reflect.Value, defaultBytes []byte, name string) reflect.V
 		runtimeDefaultable.RuntimeDefaults()
 	}
 
+	// fmt.Printf("fillDefaults on %v (%T) returning %v\n", name, obj.Interface(), obj.Interface())
```
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply master/pkg/schemas/defaults.go]: done

**Pull request merged**